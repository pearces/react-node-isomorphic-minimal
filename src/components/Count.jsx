import React, { Component } from 'react';
import { APP_NAME } from '../constants';

class Count extends Component {
  constructor(props) {
    super(props);
    this.state = { count: 0 };
  }

  clickHandler = () => this.setState(({ count }) => ({ count: count + 1 }));

  render() {
    const { count } = this.state;
    return (
      <>
        <h1>{APP_NAME}</h1>
        <div>{`The button has been clicked ${count} times.`}</div>
        <button className="primary center" type="button" label="click this" onClick={this.clickHandler}>Click this</button>
      </>
    );
  }
}

export default Count;
