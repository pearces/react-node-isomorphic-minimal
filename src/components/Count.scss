@use '../sass' as *;

@include md {
  section {
    div {
      display: flex;

      p {
        flex-basis: 25%;
      }

      button {
        flex-basis: 75%;
      }
    }
  }
}
