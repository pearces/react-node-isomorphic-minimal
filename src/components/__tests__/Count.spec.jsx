import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Count from '../Count';

describe('Count', () => {
  it('matches component snapshot', () => {
    const app = renderer.create(<Count />);
    const tree = app.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('handles button clicks', () => {
    const app = shallow(<Count />);
    app.find('button').simulate('click');
    expect(app.state('count')).toEqual(1);
  });
});
