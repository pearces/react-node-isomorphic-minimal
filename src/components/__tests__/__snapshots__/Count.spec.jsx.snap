// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Count matches component snapshot 1`] = `
Array [
  <h1>
    react-node-isomorphic-minimal app
  </h1>,
  <div>
    The button has been clicked 0 times.
  </div>,
  <button
    className="primary center"
    label="click this"
    onClick={[Function]}
    type="button"
  >
    Click this
  </button>,
]
`;
